gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/adwords/GoogleAdWordsCredential.java

```diff
@@ -0,0 +1,179 @@
+package gobblin.ingestion.google.adwords;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Properties;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.PropertiesConfiguration;
+
+import com.google.api.ads.adwords.lib.client.AdWordsSession;
+import com.google.api.ads.common.lib.auth.GoogleClientSecretsBuilder;
+import com.google.api.ads.common.lib.exception.ValidationException;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeTokenRequest;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+
+public class GoogleAdWordsCredential {
+  private static final String REDIRECT_URI = "urn:ietf:wg:oauth:2.0:oob";
+  private static final ArrayList<String> SCOPES = Lists.newArrayList("https://www.googleapis.com/auth/adwords");
```

Comment:
Maybe not necessary. This piece of code to generate the credentials will only be run locally to generate the keys. The scheduled job will never call this part.